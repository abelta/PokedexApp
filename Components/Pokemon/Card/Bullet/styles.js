import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  bullet: {
    borderColor: 'black',
    flexDirection: 'row',
    padding: 2,
  },
  label: {
    backgroundColor: 'black',
    color: 'white',
    flex: 1,
    paddingLeft: 10,
  },
  value: {
    backgroundColor: 'white',
    color: 'black',
    flex: 1,
    paddingLeft: 10,
  },
});

export default styles;
