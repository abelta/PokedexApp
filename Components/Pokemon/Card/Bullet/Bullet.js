import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import styles from './styles';

const Bullet = ({ label, style, value }) => (
  <View style={[styles.bullet, style]}>
    <Text style={styles.label}>{label}</Text>
    <Text style={styles.value}>{value}</Text>
  </View>
);

Bullet.propTypes = {
  label: PropTypes.string.isRequired,
  // eslint-disable-next-line react/require-default-props
  style: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

export default Bullet;
