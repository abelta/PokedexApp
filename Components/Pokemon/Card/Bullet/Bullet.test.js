import React from 'react';
import renderer from 'react-test-renderer';
import Bullet from '.';

describe('Bullet', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Bullet label="label" value="4" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
