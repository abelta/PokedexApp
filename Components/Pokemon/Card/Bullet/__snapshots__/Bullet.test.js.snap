// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bullet renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "red",
        "borderColor": "black",
        "flexDirection": "row",
        "height": 20,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "black",
        "color": "white",
        "flex": 1,
        "paddingLeft": 10,
      }
    }
  >
    label
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "white",
        "color": "black",
        "flex": 1,
        "paddingLeft": 10,
      }
    }
  >
    4
  </Text>
</View>
`;
